This is a trial to
Automatic Drum Transcription Library (ADTLib)
https://github.com/CarlSouthall/ADTLib
on google coloaboratory
 using  tf1.5.x cpu and python3.6.x.   
(May 2023)

(1) install python3.6.x to use tf 1.5.x

In [1]:
!sudo apt install python3.6
!sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.6 1

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  libpython3.6-minimal libpython3.6-stdlib python3.6-minimal
Suggested packages:
  python3.6-venv binfmt-support
The following NEW packages will be installed:
  libpython3.6-minimal libpython3.6-stdlib python3.6 python3.6-minimal
0 upgraded, 4 newly installed, 0 to remove and 24 not upgraded.
Need to get 4,294 kB of archives.
After this operation, 22.1 MB of additional disk space will be used.
Get:1 http://ppa.launchpad.net/deadsnakes/ppa/ubuntu focal/main amd64 libpython3.6-minimal amd64 3.6.15-1+focal3 [569 kB]
Get:2 http://ppa.launchpad.net/deadsnakes/ppa/ubuntu focal/main amd64 python3.6-minimal amd64 3.6.15-1+focal3 [1,718 kB]
Get:3 http://ppa.launchpad.net/deadsnakes/ppa/ubuntu focal/main amd64 libpython3.6-stdlib amd64 3.6.15-1+focal3 [1,758 kB]
Get:4 http://ppa.launchpad.net/deadsnakes/ppa/ubuntu focal/main amd64 python3.6 amd64 3.

(2)change python3 version python3.6. input 2 and \enter

In [2]:
!sudo update-alternatives --config python3

There are 3 choices for the alternative python3 (providing /usr/bin/python3).

  Selection    Path                 Priority   Status
------------------------------------------------------------
* 0            /usr/bin/python3.10   2         auto mode
  1            /usr/bin/python3.10   2         manual mode
  2            /usr/bin/python3.6    1         manual mode
  3            /usr/bin/python3.8    1         manual mode

Press <enter> to keep the current choice[*], or type selection number: 2
update-alternatives: using /usr/bin/python3.6 to provide /usr/bin/python3 (python3) in manual mode


(3)check current python version if Python 3.6.x

In [3]:
!python --version

Python 3.6.15


(4)install tools for python 3.6.x

In [4]:
!sudo apt install python3-pip
!sudo apt-get install python3.6-distutils

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  python-pip-whl python3-setuptools python3-wheel
Suggested packages:
  python-setuptools-doc
The following NEW packages will be installed:
  python-pip-whl python3-pip python3-setuptools python3-wheel
0 upgraded, 4 newly installed, 0 to remove and 24 not upgraded.
Need to get 2,389 kB of archives.
After this operation, 4,933 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 python-pip-whl all 20.0.2-5ubuntu1.8 [1,805 kB]
Get:2 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 python3-setuptools all 45.2.0-1ubuntu0.1 [330 kB]
Get:3 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 python3-wheel all 0.34.2-1ubuntu0.1 [23.9 kB]
Get:4 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 python3-pip all 20.0.2-5ubuntu1.8 [231 kB]
Fetched 2,389 kB in 0s (16

(5)install tf 1.5.x, cpu version

In [5]:
!pip uninstall tensorflow
!pip install tensorflow==1.15.0


Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 412.3 MB 30 kB/s 
     |████████████████████████████████| 3.8 MB 48.5 MB/s 
     |████████████████████████████████| 65 kB 3.0 MB/s 
     |████████████████████████████████| 503 kB 66.4 MB/s 
     |████████████████████████████████| 1.1 MB 62.4 MB/s 
     |████████████████████████████████| 75 kB 3.1 MB/s 
     |████████████████████████████████| 126 kB 48.9 MB/s 
     |████████████████████████████████| 42 kB 929 kB/s 
     |████████████████████████████████| 4.6 MB 65.9 MB/s 
     |████████████████████████████████| 50 kB 5.8 MB/s 
     |████████████████████████████████| 14.8 MB 76.6 MB/s 
     |████████████████████████████████| 57 kB 4.5 MB/s 
     |████████████████████████████████| 289 kB 60.6 MB/s 
     |████████████████████████████████| 97 kB 5.8 MB/s 
     |████████████████████████████████| 4.0 MB 68.1 MB/s 
  Created wheel for termcolor: filename=te

(6)set default python 3.10 auto, input 0 enter and then RESTART RUNTIME
to avoid crush when RESTART.


In [6]:
!sudo update-alternatives --config python3

There are 3 choices for the alternative python3 (providing /usr/bin/python3).

  Selection    Path                 Priority   Status
------------------------------------------------------------
  0            /usr/bin/python3.10   2         auto mode
  1            /usr/bin/python3.10   2         manual mode
* 2            /usr/bin/python3.6    1         manual mode
  3            /usr/bin/python3.8    1         manual mode

Press <enter> to keep the current choice[*], or type selection number: 0
update-alternatives: using /usr/bin/python3.10 to provide /usr/bin/python3 (python3) in auto mode


(7)After RESTART RUNTIME, change python3 version python3.6. again,  input 2 and \enter

In [1]:
!sudo update-alternatives --config python3

There are 3 choices for the alternative python3 (providing /usr/bin/python3).

  Selection    Path                 Priority   Status
------------------------------------------------------------
* 0            /usr/bin/python3.10   2         auto mode
  1            /usr/bin/python3.10   2         manual mode
  2            /usr/bin/python3.6    1         manual mode
  3            /usr/bin/python3.8    1         manual mode

Press <enter> to keep the current choice[*], or type selection number: 2
update-alternatives: using /usr/bin/python3.6 to provide /usr/bin/python3 (python3) in manual mode


(8)check current python version if Python 3.6.x

In [2]:
!python --version

Python 3.6.15


(9)install other tools for python 3.6.x

In [3]:
!pip install ez_setup
!pip install --upgrade setuptools
!pip install cython
!sudo apt install python3.6-dev

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
  Created wheel for ez-setup: filename=ez_setup-0.9-py3-none-any.whl size=11014 sha256=b67c1f7034a7709e177e1feaa158dc12057c18c21a034c51c1306efe11d325cc
  Stored in directory: /root/.cache/pip/wheels/b1/68/a0/3224e9e4903eb884e775b63b1c72d82af5ac05a6288494adb8
Successfully built ez-setup
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 952 kB 12.3 MB/s 
  Attempting uninstall: setuptools
    Found existing installation: setuptools 45.2.0
    Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
    Can't uninstall 'setuptools'. No files were found to uninstall.
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 2.0 MB 13.4 MB/s 
Reading package lists... Done
Building dependency tree

(10)install ADTLib

In [4]:
# Install
!pip install ADTLib

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 8.0 MB 267 kB/s 
     |████████████████████████████████| 25.9 MB 1.6 MB/s 
     |████████████████████████████████| 20.0 MB 1.2 MB/s 
     |████████████████████████████████| 51 kB 5.5 MB/s 
  Created wheel for ADTLib: filename=ADTLib-2.1.2-py3-none-any.whl size=8035732 sha256=67acf65d33a7dcc7815e18e146394897ce2cef248ffaaf6da167b09a7205a766
  Stored in directory: /root/.cache/pip/wheels/9a/03/98/f8afee5a8b812c292b7a0c8d42ef6de1a391f25d77d601d982
  Created wheel for madmom: filename=madmom-0.16.1-cp36-cp36m-linux_x86_64.whl size=21385392 sha256=30214187f2102c52fe424b592f18f5eee7fdab23f1461cd036ce3a0934ef9fb9
  Stored in directory: /root/.cache/pip/wheels/91/68/ab/3ce619f1688fb8da45292e9e4f9797063b2628feecd30f0d9f
  Created wheel for fpdf: filename=fpdf-1.7.2-py2.py3-none-any.whl size=40722 sha256=19fb3e504c6d786c74c6bfd69bbbe945a6c2786fbf1272d988947bb9

(11)check ADTLib and tensorflow in install list

In [5]:
!pip list

Package              Version             
-------------------- --------------------
absl-py              1.4.0               
ADTLib               2.1.2               
astor                0.8.1               
cached-property      1.5.2               
Cython               0.29.34             
dataclasses          0.8                 
dbus-python          1.2.16              
ez-setup             0.9                 
fpdf                 1.7.2               
gast                 0.2.2               
google-pasta         0.2.0               
grpcio               1.48.2              
h5py                 3.1.0               
importlib-metadata   4.8.3               
Keras-Applications   1.0.8               
Keras-Preprocessing  1.1.2               
madmom               0.16.1              
Markdown             3.3.7               
mido                 1.2.10              
numpy                1.19.5              
opt-einsum           3.3.0               
pip                  20.0.2       

(12)check tf version

In [6]:
!echo 'import tensorflow as tf' >  test1.py
!echo 'print (tf.__version__)' >>  test1.py
!python test1.py

1.15.0


(13)upload drum wav files (drum1.wav, drum2.wav, and drum3.wav) from your PC into google colaboratory current directory, Manually.

(14)check drum wav files in current directory

In [7]:
!ls

drum1.wav  drum2.wav  drum3.wav  sample_data  test1.py


(15)try ADT (Automatic Drum Transcription)


In [8]:
!ADT drum1.wav





Instructions for updating:
This class is equivalent as tf.keras.layers.LSTMCell, and will be replaced by that in Tensorflow 2.0.
Instructions for updating:
This class is equivalent as tf.keras.layers.StackedRNNCells, and will be replaced by that in Tensorflow 2.0.
Instructions for updating:
Please use `keras.layers.Bidirectional(keras.layers.RNN(cell))`, which is equivalent to this API
Instructions for updating:
Please use `keras.layers.RNN(cell)`, which is equivalent to this API
Instructions for updating:
Please use `layer.add_weight` method instead.
Instructions for updating:
Call initializer instance with the dtype argument instead of passing it to the constructor
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where
Instructions for updating:

Future major versions of TensorFlow will allow gradients to flow
into the labels input on backprop by default.

See `tf.nn.softmax_cross_entropy_with_logits_v2`.






2023-05-23 05:04:55.242595: W 

In [9]:
!ADT drum2.wav





Instructions for updating:
This class is equivalent as tf.keras.layers.LSTMCell, and will be replaced by that in Tensorflow 2.0.
Instructions for updating:
This class is equivalent as tf.keras.layers.StackedRNNCells, and will be replaced by that in Tensorflow 2.0.
Instructions for updating:
Please use `keras.layers.Bidirectional(keras.layers.RNN(cell))`, which is equivalent to this API
Instructions for updating:
Please use `keras.layers.RNN(cell)`, which is equivalent to this API
Instructions for updating:
Please use `layer.add_weight` method instead.
Instructions for updating:
Call initializer instance with the dtype argument instead of passing it to the constructor
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where
Instructions for updating:

Future major versions of TensorFlow will allow gradients to flow
into the labels input on backprop by default.

See `tf.nn.softmax_cross_entropy_with_logits_v2`.






2023-05-23 05:05:18.985250: W 

In [10]:
!ADT drum3.wav





Instructions for updating:
This class is equivalent as tf.keras.layers.LSTMCell, and will be replaced by that in Tensorflow 2.0.
Instructions for updating:
This class is equivalent as tf.keras.layers.StackedRNNCells, and will be replaced by that in Tensorflow 2.0.
Instructions for updating:
Please use `keras.layers.Bidirectional(keras.layers.RNN(cell))`, which is equivalent to this API
Instructions for updating:
Please use `keras.layers.RNN(cell)`, which is equivalent to this API
Instructions for updating:
Please use `layer.add_weight` method instead.
Instructions for updating:
Call initializer instance with the dtype argument instead of passing it to the constructor
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where
Instructions for updating:

Future major versions of TensorFlow will allow gradients to flow
into the labels input on backprop by default.

See `tf.nn.softmax_cross_entropy_with_logits_v2`.






2023-05-23 05:05:40.985490: W 

(16)check result *.pdf and *.txt

In [11]:
!ls

drum1.ADT.txt	   drum2.ADT.txt      drum3.ADT.txt	 sample_data
drum1_drumtab.pdf  drum2_drumtab.pdf  drum3_drumtab.pdf  test1.py
drum1.wav	   drum2.wav	      drum3.wav
